packages/node_modules/@webex/internal-plugin-user/src/user.js

```diff
@@ -288,6 +288,87 @@ const User = WebexPlugin.extend({
   },
 
   /**
+   * Determines if the specified user needs to signup or can signin.
+   * Triggers activation email if client credentials are used.
+   * @param {Object} params
+   * @param {string} params.email (required)
+   * @returns {Promise<Object>}
+   */
+  validate(params) {
+    return new Promise((resolve, reject) => {
+      // merge options
+      const options = {...this.config.verifyDefaults, ...params};
+
+      // validate options has email key value
+      if (!options.email) {
+        reject(new Error('`options.email` is required'));
+      }
+
+      // destructure for ease of access
+      const {services} = this.webex.internal;
+
+      // scoped globals
+      let authToken;
+
+      services.updateServices('limited', 'email', options.email)
+        // retrieved an error when updating limited services
+        .catch((e) => {
+          e.message = ['failed to update limited catalog: ', e.message]
+            .join('');
+          reject(e);
+        })
+        // successfully updated limited service catalog
+        .then(() => this.webex.credentials.getUserToken())
+        // failed to get user token
+        .catch(() => this.webex.credentials.getClientToken())
+        // failed to get client token
+        .catch((e) => {
+          e.message = ['failed to get auth token: ', e.message].join('');
+          reject(e);
+        })
+        // successfully retrieved auth token
+        .then((token) => {
+          authToken = token.toString();
+
+          return services.updateServices('signin', 'email', options.email, authToken);
+        })
+        // retrieved an error when updating signin services
+        .catch(() => Promise.resolve())
```

Comment:
I think the /activations call should go in this catch instead of resolving the promise (if it's a 404). If we get a 404 from the signin catalog, it means the user hasn't completed activation. We can double check with u2c on Monday.

Reply:
And the .then() would/should contain the user details.